private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisherTest.java

```diff
@@ -0,0 +1,74 @@
+package com.github.bazelbuild.rules_jvm_external.maven;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.artifact.repository.metadata.Versioning;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Writer;
+import org.junit.Test;
+
+public class MavenPublisherTest {
+
+  @Test
+  public void testDeserialization() throws Exception {
+    String xml =
+        "<metadata modelVersion=\"1.1.0\">\n"
+            + "    <groupId>com.mycompany.app</groupId>\n"
+            + "    <artifactId>my-app</artifactId>\n"
+            + "    <versioning>\n"
+            + "        <latest>1.0</latest>\n"
+            + "        <release>1.0</release>\n"
+            + "        <versions>\n"
+            + "            <version>1.0</version>\n"
+            + "        </versions>\n"
+            + "        <lastUpdated>20200731090423</lastUpdated>\n"
+            + "    </versioning>\n"
+            + "</metadata>";
+    Metadata metadata =
+        new MetadataXpp3Reader()
+            .read(new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8)), false);
+
+    assertEquals(metadata.getGroupId(), "com.mycompany.app");
+    assertEquals(metadata.getArtifactId(), "my-app");
+
+    Versioning versioning = metadata.getVersioning();
+    assertEquals(versioning.getLatest(), "1.0");
+    assertEquals(versioning.getLastUpdated(), "20200731090423");
+  }
+
+  @Test
+  public void testSerialization() throws Exception {
+    String expected =
```

Comment:
This is fragile. If you do want to test that the third party library is working, it'd be better to make assertions about the parsed XML that's returned, rather than the exact string.